tests/app/forms/validation/test_date_range_validator.py

```diff
@@ -1,193 +1,167 @@
+    get_date_range_check, mock_form, mock_period_from, mock_period_to
+):
+    mock_period_from.data = "2016-02-01"
+    mock_period_to.data = "2016-02-12"
+    period_min = {"days": 20}
 
-        period_to = Mock()
-        period_to.data = "2018-01"
+    date_range_check = get_date_range_check(period_min=period_min)
 
-        mock_form = Mock()
-        validator(mock_form, period_from, period_to)
+    with pytest.raises(ValidationError) as context:
+        date_range_check(mock_form, mock_period_from, mock_period_to)
 
-    def test_invalid_month_year_date_range(self):
-        validator = DateRangeCheck()
+    assert "Enter a reporting period greater than or equal to 20 days" == str(
+        context.value
+    )
 
-        period_from = Mock()
-        period_from.data = "2018-07"
 
-        period_to = Mock()
-        period_to.data = "2018-06"
+def test_date_range_with_too_large_period(
+    get_date_range_check, mock_form, mock_period_from, mock_period_to
+):
+    mock_period_from.data = "2016-02-11"
+    mock_period_to.data = "2016-03-14"
+    period_max = {"months": 1}
 
-        mock_form = Mock()
+    date_range_check = get_date_range_check(period_max=period_max)
 
-        with self.assertRaises(ValidationError) as context:
-            validator(mock_form, period_from, period_to)
+    with pytest.raises(ValidationError) as context:
+        date_range_check(mock_form, mock_period_from, mock_period_to)
 
-        self.assertEqual(error_messages["INVALID_DATE_RANGE"], str(context.exception))
+    assert "Enter a reporting period less than or equal to 1 month" == str(
+        context.value
+    )
 
-    def test_date_range_with_too_small_period(self):
-        period_min = {"days": 20}
-        validator = DateRangeCheck(period_min=period_min)
 
-        period_from = Mock()
-        period_from.data = "2016-02-01"
+def test_bespoke_message_playback(
+    get_date_range_check, mock_form, mock_period_from, mock_period_to
+):
+    mock_period_from.data = "2016-02-11"
+    mock_period_to.data = "2018-03-19"
+    message = {"DATE_PERIOD_TOO_LARGE": "Test %(max)s"}
+    period_max = {"years": 2, "months": 1, "days": 3}
 
-        period_to = Mock()
-        period_to.data = "2016-02-12"
+    date_range_check = get_date_range_check(messages=message, period_max=period_max)
 
-        mock_form = Mock()
+    with pytest.raises(ValidationError) as context:
+        date_range_check(mock_form, mock_period_from, mock_period_to)
 
-        with self.assertRaises(ValidationError) as context:
-            validator(mock_form, period_from, period_to)
+    assert "Test 2 years, 1 month, 3 days" == str(context.value)
 
-        self.assertEqual(
-            "Enter a reporting period greater than or equal to 20 days",
-            str(context.exception),
-        )
 
-    def test_date_range_with_too_large_period(self):
-        period_max = {"months": 1}
-        validator = DateRangeCheck(period_max=period_max)
+def test_message_combinations(
```

Comment:
There is a lot going on in this test, I know the existing had it, but while we are here I think we should parametrize these, which has the period and expected message as params.